### PR TITLE
[DNM] mgr/dashboard: reload mgr, if user submits new SSL keys 

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Fixes:  http://tracker.ceph.com/issues/24651

Regarding the implementation: 

1. We cannot just reload the CherryPy server, as it is not properly implemented
1. We cannot reload the dashboard module within Python, as the mgr itself is restarted when enabling any module
1. My idea: just restart the mgr. Although, I'm not sure, if it's worth it. 

TODO:
- [ ] Updates documentation

### CODE REVIEW

File: src/pybind/mgr/dashboard/tools.py [hunk 1]
```diff
@@ -764,3 +766,31 @@ def str_to_bool(val):
     if isinstance(val, bool):
         return val
     return bool(strtobool(val))
+
+
+def restart_if_running():
```
Comment (jcsp) [inline src/pybind/mgr/dashboard/tools.py]: It is probably better to send a "mgr fail" command to the monitor.  That way the monitor knows that this is something intentional, and can avoid logging a warning about an unexpected daemon restart.

(ceph-mgr daemons notice if they were killed, and respawn themselves)

File: src/pybind/mgr/dashboard/tools.py [hunk 2]
```diff
@@ -764,3 +766,31 @@ def str_to_bool(val):
...
+    """Restarts ceph-mgr iff CherryPy is serving."""
```
Comment (votdev) [inline src/pybind/mgr/dashboard/tools.py]: s/iff/if/

**Comment by sebastian-philipp** [comment]
With http://tracker.ceph.com/issues/36237 implementing a module reload should be much cleaner. Does it make sense to continue with this PR?

I kind of dislike to restart the mgr from within a Python module.

File: src/pybind/mgr/dashboard/tools.py (hunk 2 above)
Comment (tchaikov) [reply src/pybind/mgr/dashboard/tools.py]: i think it's an abbreviation of "if and only if".

**Comment by jcsp** [comment]
When the dashboard has module management, I expect it'll get mgr restarts as a side effect of changes to the list of enabled modules -- but for the SSL key changes, we'll still need the explicit mechanism for the module to request a restart, right?

**Comment by sebastian-philipp** [comment]
@jcsp as far as I can see, yes, the dashboard module need to reload itslef to reload the ssl keys.

**Comment by LenzGr** [comment]
Closing per our conversation during today's dashboard standup.

**Pull request closed without merging**